fastify/fastify-cli
# Improve ts template and bump fastify-plugin

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

This is not pretty but I don't see a way around the way tap typings are defined.
`export type Test = typeof tap['Test']['prototype'];`
Previously `Test` type would just be any due to `noImplicitAny: false` by default.

* use strict: true in tsconfig.json
* include tsc in `npm test` to check types before running tests
* use ES2018 as TS target
* update code to consistently use single quotes and have capitalized
  comments
* add @types/tap and fix typings for tests
* change exports for plugins and routes to use correct types with
  type inference
* consistently use/ignore promises returned by fastify functions
* move all source to src/ folder
* create separate tsc target for test folder and run it before tap

/cc @Ethan-Arrowood
